test/spec/openapi/refs.js

```diff
@@ -282,3 +282,46 @@ test('uses examples if has property required in body', async (t) => {
   t.ok(schema.parameters)
   t.same(schema.parameters[0].in, 'query')
 })
+
+test('external schema reference', async (t) => {
+  const fastify = Fastify()
+
+  await fastify.register(fastifySwagger, {
+    openapi: {
+      components: {
+        schemas: {
+          'external-schema-01': {
+            $id: 'external-schema-01',
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string'
+              }
+            }
+          }
+        }
+      }
+    }
+  })
+
+  fastify.get('/', {
+    schema: {
+      response: {
+        200: {
```

Comment:
You dont test your change imho. here we need something like { $ref: 'external-schema-01' }